Sources/_StringProcessing/Algorithms/Consumers/RegexConsumer.swift

```diff
@@ -0,0 +1,68 @@
+import _MatchingEngine
+
+public struct RegexConsumer<Consumed: BidirectionalCollection> where Consumed.SubSequence == Substring {
```

Comment:
Does this need to be public?

Reply:
Also used in some return types of public methods, in this case `.ranges(of: /.../)` and `.split(by: /.../)`.